src/main/java/org/jenkinsci/plugins/workflow/libs/LibrariesAction.java

```diff
@@ -39,8 +39,15 @@
 class LibrariesAction extends InvisibleAction {
 
     private final List<LibraryRecord> libraries;
-
+    
+    private final String scope;
+    
     LibrariesAction(List<LibraryRecord> libraries) {
+        this(null, libraries);
+    }
+
+    LibrariesAction(String scope, List<LibraryRecord> libraries) {
+        this.scope = scope;
```

Comment:
New constructor & field added to `LibrariesAction` to allow recording of "scope" - what source file the libraries in this LibrariesAction are from.